code/main.cpp

```diff
@@ -1,12 +1,83 @@
+/*
+ * This main is used to test the canbus or beetle class.
+ */
+
 #include <hwlib.hpp>
-#include <qik_2s12v10.hpp>
+#include <beetle.hpp>
+#include <comm.hpp>
 
 int main(void) {
     // kill the watchdog
     WDT->WDT_MR = WDT_MR_WDDIS;
     hwlib::wait_ms(1000);
 
-    while (true) {
+    // canbus beetle testing
+    bool beetle_canbus = false;
+    // beetle testing
+    bool test_set_speed = true;
+    //bool test_get_configuration_parameter = true;
+    //bool test_get_error = true;
+
+    auto qik_2s12v10_reset_pin = hwlib::target::pin_out(2, 25); // digital pin 5
+    r2d2::uart_ports_c uart_port_one = r2d2::uart_ports_c::uart1;
+
+    r2d2::comm_c comm;
+    auto beetle = r2d2::moving_platform::beetle_c(
+        uart_port_one, 9600u, qik_2s12v10_reset_pin, comm);
+
+    if(beetle_canbus){
+        while (1){
+            beetle.process();
+            hwlib::wait_ms(100);
+        }
+    }
+    
+    
+
+
+    if (test_set_speed) {
+        //motor tests:
+        hwlib::cout << "Testing both motors, 31% power forward.\n";
+        beetle.set_speed(80);
+        beetle.turn(0);
         hwlib::wait_ms(1000);
+        beetle.set_speed(0);
+
+        hwlib::wait_ms(500);
+
+        hwlib::cout << "Testing both motors, 31% power backward.\n";
+        beetle.set_speed(-80);
+        beetle.turn(0);
+        hwlib::wait_ms(2000);
+        beetle.set_speed(0);
+
+        hwlib::wait_ms(500);
+        hwlib::cout << "Testing turning 60 degrees left.\n";
+        beetle.set_speed(40);
+        beetle.turn(60);
+        hwlib::wait_ms(3000);
+        beetle.set_speed(0);
+
+
+        hwlib::wait_ms(500);
+        beetle.set_speed(0);
     }
+
+    /*if (test_get_configuration_parameter) {
```

Comment:
This code does not need to be commented.

Reply:
They don't now qik, since I replaced it with beetle code. This will be needed fro the qik test, but this is not present yet

Reply:
in the new commit it is deleted